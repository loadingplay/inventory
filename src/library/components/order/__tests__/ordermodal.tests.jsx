import React from "react";
import { OrderModal } from "..";
import { shallow } from "enzyme";


describe("OrderModal", () => {
  it("should match snap shot", () => {
    const wrapper = shallow(<OrderModal />, {disableLifecycleMethods: true});
    expect(wrapper).toMatchSnapshot();
  });

  it("should load document", async () => {
    const wrapper = shallow(<OrderModal />, {disableLifecycleMethods: true});
    wrapper.instance().order_service.get = jest.fn().mockReturnValue({
      order: {
        extra_info: JSON.stringify({dispatch_guide: {url: "test"}})
      }
    });

    await wrapper.instance().loadDocument();
    expect(wrapper.instance().state.show_guide).toBeTruthy();
    expect(wrapper.instance().state.guide_url).toEqual("test");
  });
});
