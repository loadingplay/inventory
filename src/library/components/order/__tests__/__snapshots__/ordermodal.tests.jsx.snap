// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderModal should match snap shot 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={0}
  contentLabel="Example Modal"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {
        "bottom": "auto",
        "left": "50%",
        "marginRight": "-50%",
        "right": "auto",
        "top": "50%",
        "transform": "translate(-50%, -50%)",
        "width": "400px",
        "zIndex": "10",
      },
      "overlay": Object {
        "zIndex": "10",
      },
    }
  }
>
  <button
    className="order-modal-close"
    onClick={[Function]}
  >
    X
  </button>
  <div>
    Esperando documento 
  </div>
</Modal>
`;
